nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkServiceImpl.java

```diff
@@ -105,6 +107,19 @@ public BulkStatus getStatus(String bulkId) {
         return status;
     }
 
+    @Override
+    public boolean await(String bulkId, long timeout, TimeUnit unit) throws InterruptedException {
+        long deadline = System.currentTimeMillis() + unit.toMillis(timeout);
+        KeyValueStore kvStore = getKvStore();
+        do {
+            if (COMPLETED.toString().equals(kvStore.getString(bulkId + STATE))) {
+                return true;
+            }
+            Thread.sleep(1000);
```

Comment:
Can't we have a smaller sleep to be more reactive?

Reply:
But FYI this suggests that we maybe should have, if possible, a notification mechanism in the key/value store itself, to avoid such sleeps...